aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,180 @@
+    this.dynamo = dynamo;
+    this.awsProperties = awsProperties;
+    ensureLockTableExists();
+  }
+
+  private void ensureLockTableExists() {
+    ensureTableExists(awsProperties.glueCatalogLockTable(), LOCK_TABLE_SCHEMA, LOCK_TABLE_COL_DEFINITIONS);
+  }
+
+  private void ensureTableExists(String tableName, List<KeySchemaElement> schema,
+                                 List<AttributeDefinition> definitions) {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
+    } catch (ResourceNotFoundException e) {
+      LOG.info("Glue lock DynamoDB table <{}> not found, try to create", tableName);
+      dynamo.createTable(CreateTableRequest.builder()
+          .tableName(tableName)
+          .keySchema(schema)
+          .attributeDefinitions(definitions)
+          .billingMode(BillingMode.PAY_PER_REQUEST)
+          .build());
+
+      boolean isTableActive = false;
+      while (!isTableActive) {
+        LOG.info("waiting for DynamoDB table <{}> to be active", tableName);
+        try {
+          Thread.sleep(5000);
+        } catch (InterruptedException ie) {
+          LOG.warn("Glue lock DynamoDB table creation sleep interrupted", e);
+        }
+        DescribeTableResponse describeTableResponse = dynamo.describeTable(DescribeTableRequest.builder()
+            .tableName(tableName)
+            .build());
+        isTableActive = describeTableResponse.table().tableStatus().equals(TableStatus.ACTIVE);
+      }
+    }
+  }
+
+
+  @Override
+  public boolean tryLock(String database, String table, long expireMillis) {
+    String tableId = tableId(database, table);
+
+    Map<String, AttributeValue> key = Maps.newHashMap();
+    key.put(LOCK_TABLE_COL_TABLE_ID, AttributeValue.builder().s(tableId).build());
+    key.put(LOCK_TABLE_COL_EXPIRE_TS_MILLIS, AttributeValue.builder().n(
+        Long.toString(System.currentTimeMillis() + expireMillis)
+    ).build());
+
+    Map<String, AttributeValue> expressionValues = Maps.newHashMap();
+    expressionValues.put(":tid", AttributeValue.builder().s(tableId).build());
+    expressionValues.put(":ts", AttributeValue.builder().n(Long.toString(System.currentTimeMillis())).build());
+
+    try {
+
+      dynamo.putItem(PutItemRequest.builder()
+          .tableName(awsProperties.glueCatalogLockTable())
+          .item(key)
+          // succeed only if there is no lock, or the lock is there but already expired
+          .conditionExpression("attribute_not_exists(" +
+              LOCK_TABLE_COL_TABLE_ID + ") OR (" +
+              LOCK_TABLE_COL_TABLE_ID + " = :tid AND " +
+              LOCK_TABLE_COL_EXPIRE_TS_MILLIS + " < :ts)")
+          .expressionAttributeValues(expressionValues)
+          .build());
+      return true;
+    } catch (Exception e) {
+      // most likely it's ConditionalCheckFailedException, but we will catch any exception
+      LOG.debug("Acquiring lock {}.{} failed", database, table, e);
```

Comment:
this should be warn.

Reply:
+1, personally speaking non-`ConditionalCheckFailedException` may even worth a `log.error`

Reply:
> +1, personally speaking non-`ConditionalCheckFailedException` may even worth a `log.error`

ConditionalCheckFailedException is expected, that is why I put it only in debug log. But yes I agree non-ConditionalCheckFailedException should be error.